### ISSUE TITLE
Allow removal of "Version" header in PGP encryption

### ISSUE THREAD

**Issue by boegi1** [bcgit/bc-csharp#533]
PGP encrypted armored files always contain the "Version" header, also after setting the header value to null (-> "Version: ").
Could you please change this, so that when setting the version header to null it's completely removed?

**Proposed change** (https://github.com/bcgit/bc-csharp/blob/master/crypto/src/bcpg/ArmoredOutputStream.cs):

***Current implementation***

    public ArmoredOutputStream(Stream outStream, IDictionary<string, string> headers)
                : this(outStream)
            {
                foreach (var header in headers)
                {
                    var headerList = new List<string>(1);
                    headerList.Add(header.Value);
                    m_headers[header.Key] = headerList;
                }
            }

***Solution***

	public ArmoredOutputStream(Stream outStream, IDictionary<string, string> headers)
            : this(outStream)
        {
            foreach (var header in headers)
                SetHeader(header.Key, header.Value);
        }

**Comment by cipherboy** [comment]
@boegi1 Sorry if I'm missing something, but could call `headers.Remove("Version")` instead of setting it to `null`?

**Comment by cipherboy** [comment]
@boegi1 Right, I think this is fine for after-the-fact removal. But I'm curious about why you'd want to pass a dictionary with a header key (set to the `null` value) to the constructor in the first place, versus not including the header in the constructor's header dictionary at all, _given_ you don't want them in the AOS? :-) My 2c.

**Comment by boegi1** [comment]
@cipherboy Doesn't passing an empty header Dictionary automatically add the version header?

**Comment by cipherboy** [comment]
Ah, so we do:

```c#
using Org.BouncyCastle.Bcpg;

var headers = new Dictionary<string, string>();
ArmoredOutputStream aos = new ArmoredOutputStream(Console.OpenStandardOutput(), headers);

aos.WriteByte(0x00);
aos.Close();
```

I think this change makes sense, but I'll double check with @peterdettman that he doesn't think it'll break anything. Thanks!

**Comment by peterdettman** [comment]
I chose to add constructors with an `addVersionHeader` argument instead.

**Comment by peterdettman** [comment]
Already addressed.
